www/selectsource.php

```diff
@@ -34,9 +34,43 @@
 
 if (array_key_exists('idphint', $state)) {
     $parts = explode(',', $state['idphint']);
+
     if (count($parts) === 1) {
         $state['saml:idp'] = urldecode($parts[0]);
         Campusidp::delegateAuthentication($state[Campusidp::SP_SOURCE_NAME], $state);
+    } else {
+        $idphint = [];
+        foreach ($parts as $part) {
+            $idphint[] = urldecode($part);
+        }
+
+        if (count($idphint) < 6) {
+            $ch = curl_init();
+
+            curl_setopt(
+                $ch,
+                CURLOPT_URL,
+                Module::getModuleURL('campusmultiauth/idpSearch.php?idphint=' . json_encode($idphint))
+            );
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+            $idpsAsConfigItems = json_decode(curl_exec($ch));
+            curl_close($ch);
+
+            $hintComponentConfig = [];
+            $hintComponentConfig['name'] = 'individual_identities';
+            $hintComponentConfig['priority'] = 'primary';
+            $hintComponentConfig['title'] = '';
+            $hintComponentConfig['number_shown'] = 5;
```

Comment:
```suggestion
        define('HINTED_IDPS_SHOWN', 5);
        if (count($idphint) <= HINTED_IDPS_SHOWN) {
            $ch = curl_init();

            curl_setopt(
                $ch,
                CURLOPT_URL,
                Module::getModuleURL('campusmultiauth/idpSearch.php?idphint=' . json_encode($idphint))
            );
            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

            $idpsAsConfigItems = json_decode(curl_exec($ch));
            curl_close($ch);

            $hintComponentConfig = [];
            $hintComponentConfig['name'] = 'individual_identities';
            $hintComponentConfig['priority'] = 'primary';
            $hintComponentConfig['title'] = '';
            $hintComponentConfig['number_shown'] = HINTED_IDPS_SHOWN;
```
constant for easier changes